src/game/BattleGround/BattleGroundWS.h

```diff
@@ -115,21 +124,21 @@ class BattleGroundWS : public BattleGround
         virtual void StartingEventOpenDoors() override;
 
         /* BG Flags */
-        ObjectGuid GetAllianceFlagCarrierGuid() const { return m_flagCarrierAlliance; }
-        ObjectGuid GetHordeFlagCarrierGuid() const { return m_flagCarrierHorde; }
-
-        void SetAllianceFlagCarrier(ObjectGuid guid) { m_flagCarrierAlliance = guid; }
-        void SetHordeFlagCarrier(ObjectGuid guid) { m_flagCarrierHorde = guid; }
+        ObjectGuid GetFlagCarrierGuid(uint8 teamIdx) const { return m_FlagCarrier[teamIdx]; }
+        void SetFlagCarrier(uint8 teamIdx, ObjectGuid guid) { m_FlagCarrier[teamIdx] = guid; }
 
-        void ClearAllianceFlagCarrier() { m_flagCarrierAlliance.Clear(); }
-        void ClearHordeFlagCarrier() { m_flagCarrierHorde.Clear(); }
+        void ClearFlagCarrier(uint8 teamIdx) { m_FlagCarrier[teamIdx].Clear(); }
+        void ClearAllianceFlagCarrier() { m_FlagCarrier[TEAM_INDEX_ALLIANCE].Clear(); }
+        void ClearHordeFlagCarrier() { m_FlagCarrier[TEAM_INDEX_HORDE].Clear(); }
 
-        bool IsAllianceFlagPickedUp() const { return !m_flagCarrierAlliance.IsEmpty(); }
-        bool IsHordeFlagPickedUp() const { return !m_flagCarrierHorde.IsEmpty(); }
+        bool IsFlagPickedUp(uint8 teamIdx) const { return !m_FlagCarrier[teamIdx].IsEmpty(); }
+        bool IsAllianceFlagPickedUp() const { return !m_FlagCarrier[TEAM_INDEX_ALLIANCE].IsEmpty(); }
+        bool IsHordeFlagPickedUp() const { return !m_FlagCarrier[TEAM_INDEX_HORDE].IsEmpty(); }
```

Comment:
The generic version that you added should be called instead of the team versions now